File: src/main/java/no/novari/flyt/configuration/validation/valueparsability/ValueParsabilityValidator.java
package no.novari.flyt.configuration.validation.valueparsability;

import no.novari.flyt.configuration.model.configuration.dtos.ValueMappingDto;
import no.novari.flyt.configuration.model.configuration.entities.ValueMapping;

public interface ValueParsabilityValidator {

    default boolean isValid(ValueMappingDto valueMappingDto) {
        return valueMappingDto.getType() != getTypeToValidate() ||
                valueMappingDto.getMappingString() == null ||
                isValid(valueMappingDto.getMappingString());
    }

    ValueMapping.Type getTypeToValidate();

    boolean isValid(String value);

}

File: src/main/kotlin/no/novari/flyt/configuration/validation/valueparsability/ValueParsabilityValidator.kt
package no.novari.flyt.configuration.validation.valueparsability

import no.novari.flyt.configuration.model.configuration.dtos.ValueMappingDto
import no.novari.flyt.configuration.model.configuration.entities.ValueMapping

interface ValueParsabilityValidator {
    fun isValid(valueMappingDto: ValueMappingDto): Boolean =
        valueMappingDto.type != getTypeToValidate() ||
            valueMappingDto.mappingString?.let(::isValid) != false

    fun getTypeToValidate(): ValueMapping.Type

    fun isValid(value: String): Boolean
}
